### PR TITLE
[CORE] Rename the prefix of jars from spark to xsql.

### PR DESCRIPTION
**Pull request by beliefer**
**What changes were proposed in this pull request?**
If we build XSQL, then some jars created as follows:
```
spark-catalyst_2.11.jar
spark-sql_2.11.jar
spark-hive_2.11.jar
spark-hive-thriftserver_2.11.jar
spark-xsql_2.11.jar
spark-xsql-druid_2.11.jar
spark-xsql-hbase_2.11.jar
spark-xsql-monitor_2.11.jar
spark-xsql-shell_2.11.jar
```
But there exists two issues. First, the name of some jars ( e.g. spark-catalyst_2.11.jar ) is the same as the jars released by `Apache Spark`. This issue will lead to some conflict about dependency. Second, some jars ( e.g. spark-xsql-shell_2.11.jar ) is exactly XSQL  component.

### CODE REVIEW

File: assembly/pom.xml [hunk 1]
```diff
@@ -66,9 +66,21 @@
...
+      <artifactId>spark-sql_${scala.binary.version}</artifactId>
+      <version>${spark.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-catalyst_${scala.binary.version}</artifactId>
+      <version>${spark.version}</version>
+      <scope>provided</scope>
+    </dependency>
```
Comment (beliefer) [inline assembly/pom.xml]: Why there need to add catalyst dependency?

Comment (WeiWenda) [reply assembly/pom.xml]: If we don't mark spark-sql as provided, final assembly jars will contains spark-sql as well as xsql-sql module at same time which will lead to Class conflict.
image

File: assembly/pom.xml [hunk 2]
```diff
@@ -193,19 +205,31 @@
...
+          <artifactId>xsql-hive_${scala.binary.version}</artifactId>
           <version>${project.version}</version>
         </dependency>
+        <dependency>
+          <groupId>org.apache.spark</groupId>
+          <artifactId>spark-hive_${scala.binary.version}</artifactId>
+          <version>${spark.version}</version>
+          <scope>provided</scope>
+        </dependency>
       </dependencies>
```
Comment (WeiWenda) [inline assembly/pom.xml]: Actually, spark-hive don't need to mark as provided, please remove it.

File: assembly/pom.xml [hunk 3]
```diff
@@ -193,19 +205,31 @@
...
+          <artifactId>xsql-hive-thriftserver_${scala.binary.version}</artifactId>
           <version>${project.version}</version>
         </dependency>
+        <dependency>
+          <groupId>org.apache.spark</groupId>
+          <artifactId>spark-hive-thriftserver_${scala.binary.version}</artifactId>
+          <version>${spark.version}</version>
+          <scope>provided</scope>
+        </dependency>
       </dependencies>
```
Comment (WeiWenda) [inline assembly/pom.xml]: Actually, spark-hive-thriftserver don't need to mark as provided, please remove it.

File: assembly/pom.xml (hunk 1 above)
Comment (beliefer) [reply assembly/pom.xml]: You means to mark `spark-sql` not `spark-catalyst`?

File: assembly/pom.xml (hunk 2 above)
Comment (beliefer) [reply assembly/pom.xml]: This will lead final assembly jars contains `spark-hive`  we not need.

File: assembly/pom.xml (hunk 3 above)
Comment (beliefer) [reply assembly/pom.xml]: ditto

**Comment by zhangbinzaifendou** [comment]
@beliefer @WeiWenda LGTM

**Pull request closed without merging**